rust-mode.el

```diff
@@ -590,6 +649,22 @@ the desired identifiers), but does not match type annotations \"foo::<\"."
      (,(rust-re-grab (concat "#\\!?\\[" rust-re-ident "[^]]*\\]"))
       1 font-lock-preprocessor-face keep)
 
+     ;; Builtin formatting macros
+     (,(concat (rust-re-grab (concat (regexp-opt rust-builtin-formatting-macros) "!")) (concat rust-formatting-macro-opening-re rust-start-of-string-re))
+      (1 'rust-builtin-formatting-macro-face)
+      (rust-string-interpolation-matcher
```

Comment:
No, the anchored matchers are called repeatedly until point moves outside the limit (which in this case is given by `rust-end-of-string`). `rust-string-interpolation-matcher-works`, `rust-formatting-macro-font-lock`, and `rust-write-macro-font-lock` all include tests for this.